### PR TITLE
HBASE-22261 Make use of ClusterStatusListener for async client

### PR DESCRIPTION
**Pull request by Apache9**

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 20 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ master Compile Tests _ |
| 0 | mvndep | 27 | Maven dependency ordering for branch |
| +1 | mvninstall | 264 | master passed |
| +1 | compile | 79 | master passed |
| +1 | checkstyle | 106 | master passed |
| +1 | shadedjars | 284 | branch has no errors when building our shaded downstream artifacts. |
| +1 | findbugs | 291 | master passed |
| +1 | javadoc | 55 | master passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 14 | Maven dependency ordering for patch |
| +1 | mvninstall | 260 | the patch passed |
| +1 | compile | 80 | the patch passed |
| +1 | javac | 80 | the patch passed |
| -1 | checkstyle | 76 | hbase-server: The patch generated 1 new + 0 unchanged - 0 fixed = 1 total (was 0) |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 282 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 543 | Patch does not cause any errors with Hadoop 2.7.4 or 3.0.0. |
| +1 | findbugs | 327 | the patch passed |
| +1 | javadoc | 56 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 193 | hbase-client in the patch passed. |
| +1 | unit | 7653 | hbase-server in the patch passed. |
| +1 | asflicense | 46 | The patch does not generate ASF License warnings. |
| | | 10780 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-172/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/172 |
| Optional Tests |  dupname  asflicense  javac  javadoc  unit  findbugs  shadedjars  hadoopcheck  hbaseanti  checkstyle  compile  |
| uname | Linux 4f2007279cac 4.4.0-131-generic #157~14.04.1-Ubuntu SMP Fri Jul 13 08:53:17 UTC 2018 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /testptch/patchprocess/precommit/personality/provided.sh |
| git revision | master / 353f9226c3 |
| maven | version: Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-17T18:33:14Z) |
| Default Java | 1.8.0_181 |
| findbugs | v3.1.11 |
| checkstyle | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-172/1/artifact/out/diff-checkstyle-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-172/1/testReport/ |
| Max. process+thread count | 4485 (vs. ulimit of 10000) |
| modules | C: hbase-client hbase-server U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-172/1/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 297 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ master Compile Tests _ |
| 0 | mvndep | 35 | Maven dependency ordering for branch |
| +1 | mvninstall | 267 | master passed |
| +1 | compile | 77 | master passed |
| +1 | checkstyle | 97 | master passed |
| +1 | shadedjars | 275 | branch has no errors when building our shaded downstream artifacts. |
| +1 | findbugs | 245 | master passed |
| +1 | javadoc | 56 | master passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 16 | Maven dependency ordering for patch |
| +1 | mvninstall | 243 | the patch passed |
| +1 | compile | 76 | the patch passed |
| +1 | javac | 76 | the patch passed |
| -1 | checkstyle | 67 | hbase-server: The patch generated 1 new + 0 unchanged - 0 fixed = 1 total (was 0) |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 270 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 508 | Patch does not cause any errors with Hadoop 2.7.4 or 3.0.0. |
| +1 | findbugs | 286 | the patch passed |
| +1 | javadoc | 57 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 210 | hbase-client in the patch passed. |
| -1 | unit | 14387 | hbase-server in the patch failed. |
| +1 | asflicense | 62 | The patch does not generate ASF License warnings. |
| | | 17659 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.client.TestFromClientSide3 |
|   | hadoop.hbase.client.TestFromClientSideWithCoprocessor |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-172/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/172 |
| Optional Tests |  dupname  asflicense  javac  javadoc  unit  findbugs  shadedjars  hadoopcheck  hbaseanti  checkstyle  compile  |
| uname | Linux e292ba7fbc00 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /testptch/patchprocess/precommit/personality/provided.sh |
| git revision | master / f30d6c958a |
| maven | version: Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-17T18:33:14Z) |
| Default Java | 1.8.0_181 |
| findbugs | v3.1.11 |
| checkstyle | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-172/2/artifact/out/diff-checkstyle-hbase-server.txt |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-172/2/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-172/2/testReport/ |
| Max. process+thread count | 5104 (vs. ulimit of 10000) |
| modules | C: hbase-client hbase-server U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-172/2/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

File: hbase-client/src/main/java/org/apache/hadoop/hbase/client/AsyncConnectionImpl.java [hunk 1]
```diff
@@ -133,6 +139,31 @@ public AsyncConnectionImpl(Configuration conf, AsyncRegistry registry, String cl
     } else {
       nonceGenerator = NO_NONCE_GENERATOR;
     }
+    ClusterStatusListener listener = null;
+    if (conf.getBoolean(STATUS_PUBLISHED, STATUS_PUBLISHED_DEFAULT)) {
+      // XXX: this maybe a blocking operation, better to create it outside the constructor and pass
```
Comment (infraio) [inline hbase-client/src/main/java/org/apache/hadoop/hbase/client/AsyncConnectionImpl.java]: XXX ==> TODO?

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 45 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ master Compile Tests _ |
| 0 | mvndep | 21 | Maven dependency ordering for branch |
| +1 | mvninstall | 240 | master passed |
| +1 | compile | 79 | master passed |
| +1 | checkstyle | 99 | master passed |
| +1 | shadedjars | 267 | branch has no errors when building our shaded downstream artifacts. |
| +1 | findbugs | 274 | master passed |
| +1 | javadoc | 57 | master passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 15 | Maven dependency ordering for patch |
| +1 | mvninstall | 247 | the patch passed |
| +1 | compile | 77 | the patch passed |
| +1 | javac | 77 | the patch passed |
| +1 | checkstyle | 99 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 268 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 500 | Patch does not cause any errors with Hadoop 2.7.4 or 3.0.0. |
| +1 | findbugs | 294 | the patch passed |
| +1 | javadoc | 55 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 198 | hbase-client in the patch passed. |
| -1 | unit | 14133 | hbase-server in the patch failed. |
| +1 | asflicense | 60 | The patch does not generate ASF License warnings. |
| | | 17123 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.client.TestFromClientSide3 |
|   | hadoop.hbase.client.TestFromClientSideWithCoprocessor |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-172/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/172 |
| Optional Tests |  dupname  asflicense  javac  javadoc  unit  findbugs  shadedjars  hadoopcheck  hbaseanti  checkstyle  compile  |
| uname | Linux 6ef788029a78 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /testptch/patchprocess/precommit/personality/provided.sh |
| git revision | master / f30d6c958a |
| maven | version: Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-17T18:33:14Z) |
| Default Java | 1.8.0_181 |
| findbugs | v3.1.11 |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-172/3/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-172/3/testReport/ |
| Max. process+thread count | 5112 (vs. ulimit of 10000) |
| modules | C: hbase-client hbase-server U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-172/3/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 45 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ master Compile Tests _ |
| 0 | mvndep | 26 | Maven dependency ordering for branch |
| +1 | mvninstall | 301 | master passed |
| +1 | compile | 98 | master passed |
| +1 | checkstyle | 118 | master passed |
| +1 | shadedjars | 327 | branch has no errors when building our shaded downstream artifacts. |
| +1 | findbugs | 363 | master passed |
| +1 | javadoc | 68 | master passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 17 | Maven dependency ordering for patch |
| +1 | mvninstall | 312 | the patch passed |
| +1 | compile | 98 | the patch passed |
| +1 | javac | 98 | the patch passed |
| +1 | checkstyle | 123 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 330 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 637 | Patch does not cause any errors with Hadoop 2.7.4 or 3.0.0. |
| +1 | findbugs | 387 | the patch passed |
| +1 | javadoc | 69 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 227 | hbase-client in the patch passed. |
| -1 | unit | 14533 | hbase-server in the patch failed. |
| +1 | asflicense | 58 | The patch does not generate ASF License warnings. |
| | | 18240 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.client.TestFromClientSideWithCoprocessor |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-172/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/172 |
| Optional Tests |  dupname  asflicense  javac  javadoc  unit  findbugs  shadedjars  hadoopcheck  hbaseanti  checkstyle  compile  |
| uname | Linux 662edc225e52 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /testptch/patchprocess/precommit/personality/provided.sh |
| git revision | master / 0645d5a033 |
| maven | version: Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-17T18:33:14Z) |
| Default Java | 1.8.0_181 |
| findbugs | v3.1.11 |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-172/4/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-172/4/testReport/ |
| Max. process+thread count | 4974 (vs. ulimit of 10000) |
| modules | C: hbase-client hbase-server U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-172/4/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Pull request merged**